sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/jdbc/connection/ConnectionProvider.scala

```diff
@@ -46,8 +49,13 @@ private[jdbc] object ConnectionProvider extends Logging {
           logError(s"Failed to load built in provider.", t)
       }
     }
-    // Seems duplicate but it's needed for Scala 2.13
-    providers.toSeq
+
+    // toSeq seems duplicate but it's needed for Scala 2.13
+    val sparkConf = SparkEnv.get.conf
+    providers.filter { p =>
+      val key = providerEnabledConfig.format(p.name)
+      sparkConf.getOption(key).forall(_.toBoolean)
```

Comment:
Should we maybe throw a better exception saying that the configuration only takes `true` or `false`?

Reply:
Adding more descriptive message.

Reply:
Code changed so not needed.